rust/otap-dataflow/crates/state/src/store.rs

```diff
@@ -144,80 +101,47 @@ impl ObservedStateStore {
-                    .per_core
-                    .get(&core_id)
-                    .is_none_or(|cs| cs.phase != phase);
-
-                // Update the core status
-                let _ = pipeline_status.per_core.insert(core_id, new_core_status);
-
-                // Only recalculate aggregate phase if the core's phase actually changed
-                if phase_changed {
-                    pipeline_status.phase = aggregate_pipeline_phase(
-                        pipeline_status.per_core.values().map(|c| c.phase),
-                    );
-                }
-            }
-            ObservedEvent::Heartbeat { key, core_id } => {
-                // Gracefully handle poisoned mutex and proceed
-                let mut pipelines = self.pipelines.lock().unwrap_or_else(|poisoned| {
-                    // Rational: We prefer to prioritize availability of the data plane over the
-                    // observed state store's consistency. Anyway, future heartbeats will correct
-                    // any inconsistency.
-                    log::warn!("ObservedStateStore mutex was poisoned; continuing with possibly inconsistent state");
-                    poisoned.into_inner()
-                });
-
-                let pipeline_status = pipelines.entry(key).or_insert_with(|| PipelineStatus {
-                    phase: PipelinePhase::Pending,
-                    per_core: HashMap::new(),
-                });
-
-                // For heartbeats, we only update the timestamp, not the phase
-                if let Some(core_status) = pipeline_status.per_core.get_mut(&core_id) {
-                    core_status.last_beat = now;
-                } else {
-                    let _ = pipeline_status.per_core.insert(
-                        core_id,
-                        CoreStatus {
-                            phase: PipelinePhase::Pending,
-                            last_beat: now,
-                        },
-                    );
-                }
+    #[allow(
+        clippy::print_stderr,
+        reason = "Use `eprintln!` while waiting for https://github.com/open-telemetry/otel-arrow/issues/1237."
+    )]
+    fn report(&self, observed_event: ObservedEvent) -> Result<ApplyOutcome, Error> {
+        // ToDo Event reporting see: https://github.com/open-telemetry/otel-arrow/issues/1237
+        // The code below is temporary and should be replaced with a proper event reporting
+        // mechanism (see previous todo).
+        match &observed_event.r#type {
+            EventType::Request(_) | EventType::Error(_) => {
+                eprintln!("Observed event: {observed_event:?}")
             }
+            EventType::Success(_) => { /* no console output for success events */ }
         }
+
+        let mut pipelines = self.pipelines.lock().unwrap_or_else(|poisoned| {
+            log::warn!(
+                "ObservedStateStore mutex was poisoned; continuing with possibly inconsistent state"
+            );
```

Comment:
I see we're using `log` in a few places in this PR. Should we add a tracing subscriber like `env_logger` in  `main.rs`?